sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/TransportClient.java

```diff
@@ -26,16 +33,29 @@ public void enableThroughputControl(ThroughputControlStore throughputControlStor
         if (StringUtils.isEmpty(request.requestContext.resourcePhysicalAddress)) {
             request.requestContext.resourcePhysicalAddress = physicalAddress.toString();
         }
+
         if (this.throughputControlStore != null) {
-            return this.throughputControlStore.processRequest(
-                request,
-                Mono.defer(() -> this.invokeStoreAsync(physicalAddress, request)));
+            return this.invokeStoreWithThroughputControlAsync(physicalAddress, request);
         }
 
-        return this.invokeStoreAsync(physicalAddress, request);
+        return this.invokeStoreInternalAsync(physicalAddress, request);
     }
 
     protected abstract Mono<StoreResponse> invokeStoreAsync(
         Uri physicalAddress,
         RxDocumentServiceRequest request);
+
+    private Mono<StoreResponse> invokeStoreWithThroughputControlAsync(Uri physicalAddress, RxDocumentServiceRequest request) {
+        return this.throughputControlStore.processRequest(
```

Comment:
Any specific reason we have not switched the thread in this scenario ?

Reply:
it is calling invokeStoreInternal, so if switch required, it will happen there

Reply:
nevermind it is going in common method